types/imap-simple/index.d.ts

```diff
@@ -42,6 +43,18 @@ export class ImapSimple extends NodeJS.EventEmitter {
     openBox(boxName: string, callback: (err: Error, boxName: string) => void): void;
     openBox(boxName: string): Promise<string>;
 
+    /** Create a mailbox, calling the provided callback with signature (err, boxName), or resolves the returned promise with boxName. */
+    addBox(boxName: string, callback: (err: Error, boxName: string) => void): void;
+    addBox(boxName: string): Promise<string>;
+
+    /** Delete a mailbox, calling the provided callback with signature (err, boxName), or resolves the returned promise with boxName. */
+    delBox(boxName: string, callback: (err: Error, boxName: string) => void): void;
+    delBox(boxName: string): Promise<string>;
+
+    /** Returns the full list of mailboxes (folders). Upon success, either the provided callback will be called with signature (err, boxes), or the returned promise will be resolved with boxes. Boxes is the exact object returned from the node-imap getBoxes() result. */
+    getBoxes(callback: (err: Error, boxes: any) => void): void;
+    getBoxes(): Promise<any>;
```

Comment:
Reading the docs it looks like the returned object can be typed `Imap.MailBoxes`.